### ISSUE TITLE
Robo3T fails to close connections to Mongo Atlas replica set eventually maxxing out

### ISSUE THREAD

**Issue by dave-digital** [Studio3T/robomongo#1662]
Robo3T Version: 1.3.1
OS: macOS Mojave 10.14.3

Issue: 
Number of open connections steadily rises using Robo3T connected to a Mongo Atlas replica set.
Under normal usage the number of connections sits around 35 - 40 for our current workload. When using Robo3T that number steadily increases due to the connections not being closed. This results in the current available connection cap being reached with an impact on other services being about to connect to the replica set.

This was verified by looking at the log files of the cluster and making a note of the current open connections.
db logs show hundreds of Robo3T connections being opened.. with no associated closing state
```
NETWORK  [listener] connection accepted from __redacted__:53747 #12915 (345 connections now open)
NETWORK  [conn12915] received client metadata from __redacted__:53747 conn12915: { application: { name: "robo3t" }, driver: { name: "MongoDB Internal Client", version: "4.0.5-17-gd808df2233" }, os: { type: "Windows", name: "Microsoft Windows 8", architecture: "x86_64", version: "6.2 (build 9200)" } }
ACCESS   [conn12915] Successfully authenticated as principal __redacted__ on admin from client __redacted__:53747
```
Current Solution:
Stop using Robo3T :( and it all returns back to normal

![Screen Shot 2019-07-26 at 10 18 24 am](https://user-images.githubusercontent.com/50893879/61917182-28ba6880-af8f-11e9-9b77-df076bf43500.png)

Has anyone else had this problem?

**Comment by Ziac** [comment]
Yes, I have the same problem. It never closes opened connections. It's so ridiculous.

**Comment by mithundas79** [comment]
An eye opener

**Comment by dave-digital** [comment]
As a result of this discovery we have now switched to Mongo Compass - which works without opening a multitude of connections - but feels stunted in features compared to Robo3T.... swings and roundabouts I guess.

**Comment by mithundas79** [comment]
> As a result of this discovery we have now switched to Mongo Compass - which works without opening a multitude of connections - but feels stunted in features compared to Robo3T.... swings and roundabouts I guess.

Well said sir @dave-digital .... i think that's the only option unless we go for the paid version of Studio 3T, I think Compass paid version is also good as well....

**Comment by Memogcia** [comment]
Is there a way to close the connections on a free tier?

**Comment by Bekt** [comment]
@Memogcia you can remove the IP from the whitelist (where robo connections are coming from) and add back after a few moments. This will clean up old open connections.

**Comment by Memogcia** [comment]
Thanks!!! it works!

**Comment by isaelfeliciano** [comment]
@Bekt I tried that once and worked but now it happened again, this time remove the IP from the whitelist is not working

**Comment by YusufMavzer** [comment]
When will this issue be fixed?

**Comment by gradinarot** [comment]
any news on this issue?

**Comment by uda** [comment]
This seems to be the result of Robo3T being a shell centric app, each tab opening a new connection (and 2 or more depending on connection settings), so on really low tiers this can very fast exhaust the limit.

For now a workaround should be to keeping the number of open tabs to a necessary minimum, take in account that if you reach the maximum and your production app is trying to create a new connection it will fail and your users will get a bad experience.

A real solution will be to share the mongo connection between tabs (except for shell tabs), it's unreasonable to have 10 open connections just to query five collections.

**Comment by simsekgokhan** [comment]
Hi @dave-digital, all. Thanks for reporting the problem with details and sorry answering so late.
We have never seen this problem during development or QA. We did a quick test on macOS Mojave with Mongo Atlas (4.2) but connections were closed as we closed tabs and finally the program.

Can anyone help how to reproduce this problem?

![image](https://user-images.githubusercontent.com/19390749/88284793-ade97200-ccf6-11ea-904d-dbbe8e70fa6f.png)
